test/sol5/kyberMatchingEngine.js

```diff
@@ -1663,6 +1663,56 @@ contract('KyberMatchingEngine', function(accounts) {
                 );
             });
 
+            describe(`should return zero rates and amounts if duplicate reserve ID is used for split`, async() => {
```

Comment:
should add hint handler test as well.

Reply:
added

Reply:
Done. Also, helped to "re-write" the tests, cos I started fixing 1 small thing that led to fixing another small thing, that led to fixing a big thing.

Just like this: https://www.youtube.com/watch?v=AbSehcT19u0